orderer/common/multichannel/registrar.go

```diff
@@ -357,14 +360,36 @@ func (r *Registrar) newChain(configtx *cb.Envelope) {
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
+	channelName, err := channelNameFromConfigTx(configtx)
+	if err != nil {
+		logger.Warnf("Failed extracting channel name: %v", err)
+		return
+	}
+
+	// fixes https://github.com/hyperledger/fabric/issues/2931
+	if existingChain, exists := r.chains[channelName]; exists {
+		if raftChain, isRaftChain := existingChain.Chain.(RaftChain); isRaftChain && raftChain.IsRaft() {
+			logger.Infof("Channel %s already created, skipping its creation", channelName)
+			return
+		}
+	}
+
+	cs := r.createNewChain(configtx)
+	cs.start()
+	logger.Infof("Created and started new channel %s", cs.ChannelID())
+}
+
+func (r *Registrar) createNewChain(configtx *cb.Envelope) *ChainSupport {
```

Comment:
I don't understand why we are creating a new function and moving the line where we start the chain?

Reply:
I thought it was a good idea to have the same methods as in 2.3 since they do not introduce any additional changes. 
Already reverted and I fixed it.  Also, I added error handling for call of ` ledgerResources.Append` from 2.3.